Modules/_zoneinfo.c

```diff
@@ -7,6 +7,17 @@
 
 #include "datetime.h"
 
+// This takes advantage of GCC compiler extensions to determine the type of the
+// variable, so we'll make Py_ASSERT_VAR_UNSIGNED a noop on non-GCC platforms.
+// This is not perfect, but it's better than nothing.
+#ifdef __GNUC__
+#define IS_TYPE_UNSIGNED(type) (((type)-1) > (type)0)
+#define Py_ASSERT_VAR_UNSIGNED(var) \
```

Comment:
We could, theoretically, move this into `pymacro.h` (or some private variant thereof), and potentially we could make it more portable, there might be an MSVC version of `__typeof__` that I'm unaware of.

If we ever start allowing C11 (under flags or whatever so that it doesn't break non-C99 compilers), I think there are ways to accomplish the same thing using just the standard.